src/Logging.AzureAppServices/src/BlobLoggerProvider.cs

```diff
@@ -63,7 +60,11 @@ internal override async Task WriteMessagesAsync(IEnumerable<LogMessage> messages
             foreach (var eventGroup in eventGroups)
             {
                 var key = eventGroup.Key;
-                var blobName = $"{_appName}/{key.Year}/{key.Month:00}/{key.Day:00}/{key.Hour:00}/{_fileName}";
+                var options = _options.CurrentValue;
+                string blobName = options.FileNameFormat(new AzureBlobLoggerContext(
+                    options.ApplicationName,
+                    options.ApplicationInstanceId + "_" + options.BlobName,
```

Comment:
nit: this should be lifted out of the loop as it wont change based on `eventGroups` and we don't need to allocate it every iteration.